src/Pack.js

```diff
@@ -0,0 +1,194 @@
+
+    };
+    this._pack = pack();
+    this._packOpacity = 0.25;
+    this._shape = constant("Circle");
+    this._shapeConfig = assign(this._shapeConfig, {
+      Circle: {
+        opacity: d => d.__d3plusOpacity__ || 1,
+        labelConfig: {
+          fontResize: true
+        }
+      }
+    });
+    this._sort = (a, b) => b.value - a.value;
+    this._sum = accessor("value");
+    this._tooltip = false;
+
+  }
+
+  /**
+      Extends the draw behavior of the abstract Viz class.
+      @private
+  */
+  _draw(callback) {
+    super._draw(callback);
+
+    const height = this._height - this._margin.top - this._margin.bottom,
+          width = this._width - this._margin.left - this._margin.right;
+
+    const diameter = Math.min(height, width);
+    const transform = `translate(${(width - diameter) / 2}, ${(height - diameter) / 2})`;
+
+    let nestedData = nest();
+    for (let i = 0; i <= this._drawDepth; i++) nestedData.key(this._groupBy[i]);
+    nestedData = nestedData.entries(this._filteredData);
+
+    const packData = this._pack
+      .padding(this._layoutPadding)
+      .size([diameter, diameter])
+      (hierarchy({key: nestedData.key, values: nestedData}, d => d.values).sum(this._sum).sort(this._sort))
+      .descendants();
+
+    packData.forEach((d, i) => {
+      d.__d3plus__ = true;
+      d.children = d.children;
+      d.data.__d3plusTooltip__ = !d.height ? true : false;
+      d.depth = d.depth;
+      d.i = i;
+      d.id = d.parent ? d.parent.data.key : null;
+      d.parent = d.parent;
+      d.x = d.x;
+      d.y = d.y;
+      if (d.height) d.data.__d3plusOpacity__ = this._packOpacity;
+    });
+
+    this._shapes.push(
+      new shapes.Circle()
+        .data(packData)
+        .select(
+          elem("g.d3plus-Pack-circle", {
+            parent: this._select,
+            enter: {transform},
+            update: {transform}
+          }).node()
+        )
+        .config({
+          label: d => d.parent && !d.children ? d.id : false
+        })
+        .config(configPrep.bind(this)(this._shapeConfig, "shape", "Circle"))
+        .render()
+    );
+
+    return this;
+  }
+
+  /**
+      @memberof Pack
+      @desc If *value* is specified, sets the hover method to the specified function and returns the current class instance.
+      @param {Function} [*value*]
+      @chainable
+   */
+  hover(_) {
+    this._hover = _;
+    this._shapes.forEach(s => s.hover(_));
+
+    if (this._legend) this._legendClass.hover(_);
+    return this;
+  }
+
+  /**
+      @memberof Pack
+      @desc If *value* is specified, sets the opacity accessor to the specified function or number and returns the current class instance. If *value* is not specified, returns the current pack opacity accessor.
+      @param {Function|Number} [*value*]
+  */
+  layoutPadding(_) {
+    return arguments.length ? (this._layoutPadding = typeof _ === "function" ? _ : constant(_), this) : this._layoutPadding;
```

Comment:
Since the internal d3 function accepts a number or a function, we don't need to wrap the user-passed value in `constant`. This line can become:

```js
return arguments.length ? (this._layoutPadding = _, this) : this._layoutPadding;
```